src/Controllers/SearchController.cs

```diff
@@ -0,0 +1,59 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using ModelRepoBrowser.Models;
+
+namespace ModelRepoBrowser.Controllers;
+
+[ApiController]
+[Route("[controller]")]
+public class SearchController : ControllerBase
+{
+    private readonly ILogger<SearchController> logger;
+
+    private readonly RepoBrowserContext context;
+
+    public SearchController(ILogger<SearchController> logger, RepoBrowserContext context)
+    {
+        this.logger = logger;
+        this.context = context;
+    }
+
+    [HttpGet]
+    public async Task<IEnumerable<Model>> Get(string q)
+    {
+        logger.LogInformation("Search with query <{SearchQuery}>", q);
+
+        context.SearchQueries.Add(new Models.SearchQuery() { Query = q });
+        context.SaveChanges();
```

Comment:
REQ: Das Loggen der Suchabfrage sollte in _Fire and Foget_-Manier geschehen, allfällige Fehler beim Loggen sollten die Suche nicht beeinträchtigen.

Beispiel:
```csharp
        _ = Task.Run(() =>
        {
            context.SearchQueries.Add(new() { Query = q });
            context.SaveChanges();
        });
```

Reply:
context ist nicht thread-safe, daher habe ich ein try-catch darum gemacht, damit ein allfälliger fehler die Suche nicht beeinträchtigt.

Reply:
Die Lösung wäre für das Loggen einen neuen Context im Task zu erstellen. So beeinträchtig das Loggen die Performance der Suche nicht. Vielleicht ist das aber alles auch kein Problem und die Suche performt wie sie soll; das kannst du besser beurteilen :-).

Reply:
Soweit ich das beurteilen kann ist die Performance im Moment ok, ich lasse es daher beim synchronen loggen mit exception handling.